pkg/controller/core/clusterqueue_controller.go

```diff
@@ -36,7 +36,9 @@ import (
 	"sigs.k8s.io/kueue/pkg/queue"
 )
 
-const wlUpdateChBuffer = 10
+type ClusterQueueUpdateWatcher interface {
+	NotifyClusterQueueUpdate(*kueue.ClusterQueue)
```

Comment:
I think it's better to make a distinction here between Update and Delete. Or you could accept two arguments old and new object (nil for Delete).

Then, in the implementation of the ResourceFlavorReconciler you can discard those updates where the ClusterQueue.Spec didn't change.

This is important because ClusterQueue has a status update for every admitted workload. And verifying if a ResourceFlavor is in use is currently expensive (a doubly nested for-loop, with a lock to the entire cache).